server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -188,17 +188,17 @@ static TransportVersion def(int id) {
      * To add a new transport version, add a new constant at the bottom of the list, above this comment. Don't add other lines,
      * comments, etc. The version id has the following layout:
      *
-     * M_NNN_SS_P
+     * M_NNN_S_PP
      *
      * M - The major version of Elasticsearch
      * NNN - The server version part
-     * SS - The serverless version part. It should always be 00 here, it is used by serverless only.
-     * P - The patch version part
+     * S - The serverless version part. It should always be 0 here, it is used by serverless only.
```

Comment:
Very optional/unrelated: I like how this is phrased before "in subsidiary repos". I know that everyone knows about serverless now, I think it would be good to keep it neutral and consistent.